### DISCUSSION TITLE
The API version 'V2' has been deprecated.

### DISCUSSION THREAD

**Discussion by bbenami** [O365/python-o365#1106] [General]
A few days ago, the code below, which had been running for several months, suddenly stopped working and returned the following error message: "The API version 'V2' has been deprecated." The error is being triggered by the get_folders() function.

```
    mailbox = account.mailbox(resource='mark@contoso.com')
    folders = mailbox.get_folders()
    inbox = mailbox.inbox_folder()
```

Any ideas how to solve this?

**Comment by alejcas** [comment]
I think you are using the Outlook protocol instead of the default MsGraph one?

**Reply by bbenami** [reply]
```
    credentials = ('xxxxx', 'xxxxx')
    account = Account(credentials, auth_flow_type='credentials', tenant_id='xxxx',
                      protocol=MSOffice365Protocol)

    if account.authenticate():
        print('Authenticated!')

    mailbox = account.mailbox(resource='yossi@univonix.com')

    folders = mailbox.get_folders()
    inbox = mailbox.inbox_folder()
```

This is how I initializing the connection. Do I need to choose a different protocol?

**Reply by alejcas** [reply]
MSOffice365Protocol was deprecated on march 2024
You will need to change this to use MsGraphProtocol.

I think you will need to authenticate again
